### ISSUE TITLE
[REVIEW]: rdataretriever: An R package for downloading, cleaning, and installing publicly available datasets

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#2800]
**Submitting author:** @henrykironde (<a href="http://orcid.org/0000-0001-7105-5808">henry senyondo</a>)
**Repository:** <a href="https://github.com/ropensci/rdataretriever" target ="_blank">https://github.com/ropensci/rdataretriever</a>
**Version:** v3.0.1
**Editor:** @fboehm
**Reviewer:** @rmhogervorst, @jsgalan
**Archive:** <a href="https://doi.org/10.5281/zenodo.4314115" target="_blank">10.5281/zenodo.4314115</a>

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

## Status

[![status](https://joss.theoj.org/papers/1a121f31d1ef48bcef1c971539888ea0/status.svg)](https://joss.theoj.org/papers/1a121f31d1ef48bcef1c971539888ea0)

Status badge code:

```
HTML: <a href="https://joss.theoj.org/papers/1a121f31d1ef48bcef1c971539888ea0"><img src="https://joss.theoj.org/papers/1a121f31d1ef48bcef1c971539888ea0/status.svg"></a>
Markdown: [![status](https://joss.theoj.org/papers/1a121f31d1ef48bcef1c971539888ea0/status.svg)](https://joss.theoj.org/papers/1a121f31d1ef48bcef1c971539888ea0)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/ropensci/rdataretriever" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@rmhogervorst & @jsgalan, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @fboehm know.

✨ **Please start on your review when you are able, and be sure to complete your review in the next six weeks, at the very latest** ✨ 

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/ropensci/rdataretriever">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@henrykironde) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/ropensci/rdataretriever">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@henrykironde) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @rmhogervorst, @jsgalan it looks like you're currently assigned to review this paper :tada:.

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1371/journal.pone.0065848 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02800/joss.02800/10.21105.joss.02800.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02800/joss.02800/10.21105.joss.02800.pdf) :page_facing_up: :point_left:

**Comment by fboehm** [comment]
@jsgalan and @RMHogervorst - please check boxes above as you complete your review. For any boxes that you can't check right now, due to needed edits or fixes, please discuss the issues here or open issues in the submission repository. And please let me know if you have any questions. Thank you again!

**Comment by RMHogervorst** [comment]
@henrykironde some notes about the JOSS paper:

The description in `Statement of need` is really beautiful and only after reading that part I finally understood what the problem is that this package is solving. I really really wish you'd include something like that in the Readme (and maybe Description) too. 


REQUIRED:
- Add citations so that the references work in the text. 
- I am asked to review a `State of the field: Do the authors describe how this software compares to other commonly-used packages?` However I don't see that in the paper at this moment. Could you add something about the state of the field? I don't know what the state of the art is at this moment for the field, but I imagine workflow tools like Drake are used? What is the current state of automation/datapreparation? Is everyone using home brewed scripts? I am genuinely curious.



OPTIONAL / SUGGESTIONS:
- I am not very knowledgeable of the world of ecology or environmental sciences but the concept of frictionless data was new for me. Maybe add one sentence about what that entails?

**Comment by RMHogervorst** [comment]
Required Issues opened 
- [ ] https://github.com/ropensci/rdataretriever/issues/260  "vignette incomplete"
- [x] https://github.com/ropensci/rdataretriever/issues/257  "clarify OSI licence"

Optional issues opened:
- [ ] https://github.com/ropensci/rdataretriever/issues/259 "suggestion for remotes in stead of devtools::install_github"

**Comment by whedon** [comment]
:wave: @rmhogervorst, please update us on how your review is going.

**Comment by whedon** [comment]
:wave: @jsgalan, please update us on how your review is going.

**Comment by fboehm** [comment]
@jsgalan - I hope that the review is going well. As you work through the checklist above, please check the boxes to reflect your approval of the checklist items. Please let me know if there's something that I might do to assist you. Thanks again!

**Comment by henrykironde** [comment]
Just an update, @RMHogervorst made some great suggestions and I am working to resolve those. Thanks to the issues btw, I guess @jsgalan is inline with the issues raised.

**Comment by fboehm** [comment]
Thank you for the update, @henrykironde ! Please let me know how the resolution of issues goes, and don't hesitate to contact me when I might help with something.

**Comment by RMHogervorst** [comment]
Just an update: I feel @henrykironde and others are doing great work. I really look forward to the finished version!

**Comment by henrykironde** [comment]
Thanks folks. We have covered all the issues, I am happy to get more recommendations to consider.

**Comment by jsgalan** [comment]
Hi all, is there any confirmation I might be missing?

I am unable to check the ticks for the revision (Just checked in two browser to discard that issue).

Thanks

**Comment by jsgalan** [comment]
Hi all,

No problem in the Conda installation

![Screen Shot 2020-11-21 at 10 37 20 AM](https://user-images.githubusercontent.com/971958/99881243-98be1a80-2be6-11eb-9136-1749a2031549.png)

But a problem was found in the R installation.

![Screen Shot 2020-11-21 at 10 43 18 AM](https://user-images.githubusercontent.com/971958/99881246-9b207480-2be6-11eb-88d6-31cb672c44df.png)

Did not found any description on the repository, are there any set requirements for the R installation?

Best

**Comment by jsgalan** [comment]
Hi all, It seems I was not in a top level repository.

![Screen Shot 2020-11-21 at 11 12 33 AM](https://user-images.githubusercontent.com/971958/99881944-ed639480-2bea-11eb-9d84-f141cb731b9b.png)

All datasets are visible from Python 

![Screen Shot 2020-11-21 at 11 13 19 AM](https://user-images.githubusercontent.com/971958/99881954-f81e2980-2bea-11eb-965b-6521681d2e06.png)

**Comment by jsgalan** [comment]
Hi all,

I had problems installing the  spatial datasets

![Screen Shot 2020-11-21 at 1 19 29 PM](https://user-images.githubusercontent.com/971958/99884442-4ab41180-2bfc-11eb-908d-bb8b8df31ccb.png)

Even though I have everything set up in Postgres

![Screen Shot 2020-11-21 at 1 16 41 PM](https://user-images.githubusercontent.com/971958/99884450-56073d00-2bfc-11eb-84d5-6e213bc073cc.png)

Any ideas?

Best

**Comment by henrykironde** [comment]
Thanks @jsgalan for testing this. Make sure you are creating the extensions in the same database that you want to install the data. The default is postgres. In case you, are using a different database, please  try to provide the database name in the function call.
Let me know if that works

**Comment by jsgalan** [comment]
@whedon assign @jsgalan

**Comment by whedon** [comment]
I'm sorry human, I don't understand that. You can see what commands I support by typing:

```
@whedon commands
```

**Comment by jsgalan** [comment]
@whedon commands

**Comment by whedon** [comment]
Here are some things you can ask me to do:

```
# List Whedon's capabilities
@whedon commands

# List of editor GitHub usernames
@whedon list editors

# List of reviewers together with programming language preferences and domain expertise
@whedon list reviewers

EDITORIAL TASKS

# Compile the paper
@whedon generate pdf

# Compile the paper from alternative branch
@whedon generate pdf from branch custom-branch-name

# Ask Whedon to check the references for missing DOIs
@whedon check references

# Ask Whedon to check repository statistics for the submitted software
@whedon check repository

```

**Comment by jsgalan** [comment]
@whedon list reviewers

**Comment by whedon** [comment]
Here's the current list of reviewers: https://bit.ly/joss-reviewers

**Comment by jsgalan** [comment]
> Thanks @jsgalan for testing this. Make sure you are creating the extensions in the same database that you want to install the data. The default is postgres. In case you, are using a different database, please try to provide the database name in the function call.
> Let me know if that works

Hi all, the extensions are created in all databases. Now I got a problem with finding shp2pgsql

<img width="507" alt="Screen Shot 2020-11-22 at 12 01 13 PM" src="https://user-images.githubusercontent.com/971958/99910217-c379a300-2cba-11eb-9108-ed0a9158e3c2.png">


 even thought the bin is listed in the $PATH

<img width="734" alt="Screen Shot 2020-11-22 at 12 03 21 PM" src="https://user-images.githubusercontent.com/971958/99910202-bd83c200-2cba-11eb-80ad-42ad572c8a01.png">

**Comment by jsgalan** [comment]
@fboehm I am unable to click any of the ticks, and I somehow managed to exclude myself from the reviewers :(

Any way this could change?

**Comment by jsgalan** [comment]
Hi @henrykironde 

Why are there no references in the article proof?

<img width="777" alt="Screen Shot 2020-11-22 at 12 05 45 PM" src="https://user-images.githubusercontent.com/971958/99910308-31be6580-2cbb-11eb-8d05-7809969a9a92.png">

Am I reading the latest version?

Best

**Comment by henrykironde** [comment]
> even thought the bin is listed in the $PATH

Can you try to run `psql` and or `shp2pgsql` from command line

About the citation, I will update that to render correctly

**Comment by jsgalan** [comment]
> > even thought the bin is listed in the $PATH
> 
> Can you try to run `psql` and or `shp2pgsql` from command line
> 
They are accesible from the command line

<img width="432" alt="Screen Shot 2020-11-22 at 12 11 46 PM" src="https://user-images.githubusercontent.com/971958/99910758-d93c9780-2cbd-11eb-933e-69624e33b3ce.png">

**Comment by fboehm** [comment]
> @fboehm I am unable to click any of the ticks, and I somehow managed to exclude myself from the reviewers :(
> 
> Any way this could change?

Yes, Let me fix that @jsgalan.

**Comment by fboehm** [comment]
@whedon reinvite @jsgalan as reviewer

**Comment by whedon** [comment]
I'm sorry human, I don't understand that. You can see what commands I support by typing:

```
@whedon commands
```

**Comment by fboehm** [comment]
@whedon commands

**Comment by whedon** [comment]
Here are some things you can ask me to do:

```
# List all of Whedon's capabilities
@whedon commands

# Assign a GitHub user as the sole reviewer of this submission
@whedon assign @username as reviewer

# Add a GitHub user to the reviewers of this submission
@whedon add @username as reviewer

# Re-invite a reviewer (if they can't update checklists)
@whedon re-invite @username as reviewer

# Remove a GitHub user from the reviewers of this submission
@whedon remove @username as reviewer

# List of editor GitHub usernames
@whedon list editors

# List of reviewers together with programming language preferences and domain expertise
@whedon list reviewers

# Change editorial assignment
@whedon assign @username as editor

# Set the software archive DOI at the top of the issue e.g.
@whedon set 10.0000/zenodo.00000 as archive

# Set the software version at the top of the issue e.g.
@whedon set v1.0.1 as version

# Open the review issue
@whedon start review

EDITORIAL TASKS

# All commands can be run on a non-default branch, to do this pass a custom 
# branch name by following the command with `from branch custom-branch-name`.
# For example:

# Compile the paper
@whedon generate pdf

# Compile the paper from alternative branch
@whedon generate pdf from branch custom-branch-name

# Remind an author or reviewer to return to a review after a
# certain period of time (supported units days and weeks)
@whedon remind @reviewer in 2 weeks

# Ask Whedon to do a dry run of accepting the paper and depositing with Crossref
@whedon accept

# Ask Whedon to check the references for missing DOIs
@whedon check references

# Ask Whedon to check repository statistics for the submitted software
@whedon check repository

EiC TASKS

# Invite an editor to edit a submission (sending them an email)
@whedon invite @editor as editor

# Reject a paper
@whedon reject

# Withdraw a paper
@whedon withdraw

# Ask Whedon to actually accept the paper and deposit with Crossref
@whedon accept deposit=true

```

**Comment by fboehm** [comment]
@whedon re-invite @jsgalan as reviewer

**Comment by whedon** [comment]
OK, the reviewer has been re-invited.

@jsgalan please accept the invite by clicking this link: https://github.com/openjournals/joss-reviews/invitations

**Comment by fboehm** [comment]
@jsgalan - sorry for the difficulty. Please click on the link in the previous comment to confirm your addition as a reviewer. After doing that, you should be able to check the boxes for the review, once you're comfortable doing so.

**Comment by jsgalan** [comment]
> @jsgalan - sorry for the difficulty. Please click on the link in the previous comment to confirm your addition as a reviewer. After doing that, you should be able to check the boxes for the review, once you're comfortable doing so.

Thanks @fboehm  we will be done with the revision soon

**Comment by henrykironde** [comment]
@jsgalan, I see you are using a Conda environment, make sure that R has access to the same Python paths as well. 
Usually, changing to a virtual environment may change the paths. R may be communicating with another python rather that where the programs are installed. 

Check the paths in R
```R
library(reticulate)
sys <- import("sys")
sys$path
#Or  
reticulate::py_config()
```
Compare to the ones when python is used from the terminal 
```Python
import sys
sys.path
```

**Comment by henrykironde** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02800/joss.02800/10.21105.joss.02800.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02800/joss.02800/10.21105.joss.02800.pdf) :page_facing_up: :point_left:

**Comment by jsgalan** [comment]
> @jsgalan, I see you are using a Conda environment, make sure that R has access to the same Python paths as well.
> Usually, changing to a virtual environment may change the paths. R may be communicating with another python rather that where the programs are installed.
> 
> Check the paths in R
> 
> ```r
> library(reticulate)
> sys <- import("sys")
> sys$path
> #Or  
> reticulate::py_config()
> ```
> 
> Compare to the ones when python is used from the terminal
> 
> ```python
> import sys
> sys.path
> ```

Hi that is pointing to the same place

<img width="691" alt="Screen Shot 2020-11-23 at 9 07 11 AM" src="https://user-images.githubusercontent.com/971958/99971641-79042f00-2d6b-11eb-8122-9c0b5eb21c0b.png">

and

<img width="751" alt="Screen Shot 2020-11-23 at 9 07 32 AM" src="https://user-images.githubusercontent.com/971958/99971666-81f50080-2d6b-11eb-9deb-bd519e640d3f.png">

**Comment by henrykironde** [comment]
`reticulate::py_config()` can we try this too

**Comment by jsgalan** [comment]
> `reticulate::py_config()` can we try this too

<img width="774" alt="Screen Shot 2020-11-23 at 9 21 05 AM" src="https://user-images.githubusercontent.com/971958/99972984-40fdeb80-2d6d-11eb-98b5-ee734fb55f46.png">

**Comment by henrykironde** [comment]
Could you please try this in the terminal
```
import subprocess
subprocess.call('shp2pgsql -d ', shell=True)
```
This must be a path problem but it is hard for me to replicate on my side. Could you try running it on another machine or in a virtual environment.

**Comment by jsgalan** [comment]
> Could you please try this in the terminal
> 
> ```
> import subprocess
> subprocess.call('shp2pgsql -d ', shell=True)
> ```
> 
> This must be a path problem but it is hard for me to replicate on my side. Could you try running it on another machine or in a virtual environment.

Hi, 


<img width="413" alt="Screen Shot 2020-11-24 at 12 04 08 PM" src="https://user-images.githubusercontent.com/971958/100127532-3450c480-2e4d-11eb-853f-5c44a1f9a24d.png">

I think it is doing exactly what I am telling it to do .. 👍 

I am finding a way to change it from /sh to /bash where everything is set up. That will probably solve it

Best

**Comment by RMHogervorst** [comment]
@fboehm All my concerns where addressed and the article proof seems in order too (I would advise another read through before publishing, because I think some words are missing, but I'm not native speaker). 
For me the review is done. It looks great, and you have my approval.

Anything else I need to do?

**Comment by fboehm** [comment]
@RMHogervorst - Thank you so much for your work on this review! There is nothing left for you to do. I'll proofread the article before publishing.

**Comment by fboehm** [comment]
@jsgalan - Thanks for your work on this review! Do you have any remaining concerns? or have they all been resolved satisfactorily?

**Comment by fboehm** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02800/joss.02800/10.21105.joss.02800.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02800/joss.02800/10.21105.joss.02800.pdf) :page_facing_up: :point_left:

**Comment by fboehm** [comment]
@henrykironde - I've opened an issue to address the citations in paper.md. https://github.com/ropensci/rdataretriever/issues/274

Once that is fixed, I'll re-generate the pdf and we can continue.

**Comment by jsgalan** [comment]
The written file needs improvement, Please check the highlighted parts below.

[10.21105.joss.02800-3.pdf](https://github.com/openjournals/joss-reviews/files/5604383/10.21105.joss.02800-3.pdf)

**Comment by jsgalan** [comment]
Hi all,

When I get to Installing Spatial Datasets I still get error

![Screen Shot 2020-11-29 at 8 36 36 PM](https://user-images.githubusercontent.com/971958/100560269-4f736800-3283-11eb-82ef-d8b4173756be.png)


But it seems that it actually works since it says in the DB schemas
![Screen Shot 2020-11-29 at 8 36 47 PM](https://user-images.githubusercontent.com/971958/100560292-5e5a1a80-3283-11eb-8ad0-9eaa09011609.png)

I confirmed  it was added on Postgres DB

![Screen Shot 2020-11-29 at 8 36 57 PM](https://user-images.githubusercontent.com/971958/100560228-2fdc3f80-3283-11eb-8447-39fd0caa075e.png)

To put my individual issues to rest,  will ask the authors @henrykironde to provide at least a working example of how this connection to the database, and some lines on how to use/show the dataset 

I am happy with rest, granted the changes to the article are made.

Best,

**Comment by henrykironde** [comment]
@jsgalan looks like there may be a problem with the version of python you are using. `psql` is trying to connect using port `5432.0` This should be `5432`. You can try to provide the arguments and see it that will work
`rdataretriever::install_postgres(dataset = "portal", user = "postgres", host = "localhost", port = 5432, database = "postgres")`

**Comment by henrykironde** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02800/joss.02800/10.21105.joss.02800.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02800/joss.02800/10.21105.joss.02800.pdf) :page_facing_up: :point_left:

**Comment by fboehm** [comment]
@jsgalan - did you resolve the issues that you raised in your last post? Do you have any remaining comments for the authors?

**Comment by jsgalan** [comment]
> @jsgalan looks like there may be a problem with the version of python you are using. `psql` is trying to connect using port `5432.0` This should be `5432`. You can try to provide the arguments and see it that will work
> `rdataretriever::install_postgres(dataset = "portal", user = "postgres", host = "localhost", port = 5432, database = "postgres")`

Hi all, this worked like a charm!

<img width="586" alt="Screen Shot 2020-12-03 at 1 29 48 PM" src="https://user-images.githubusercontent.com/971958/101072656-5a244a80-356c-11eb-835e-fbaea7158efd.png">

But, I do not know how to use it, It will be cool to add a small example on how to use this dataset in the **Installing Spatial Datasets** section

Best,

**Comment by jsgalan** [comment]
> @jsgalan - did you resolve the issues that you raised in your last post? Do you have any remaining comments for the authors?

Hi It worked, and the article seems fine.

I have no more comments, I am happy to conclude this review

Thanks

**Comment by fboehm** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02800/joss.02800/10.21105.joss.02800.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02800/joss.02800/10.21105.joss.02800.pdf) :page_facing_up: :point_left:

**Comment by fboehm** [comment]
@henrykironde - It looks like there are a couple small typos in the paper.md. I think that you should say "A Docker-based testing system" in place of "A Docker based testing systems". That is, change "systems" to "system" and insert a hyphen between "Docker" and "based". Once you make these fixes, we can proceed.

**Comment by henrykironde** [comment]
Covered, let me know in case I missed something.

**Comment by fboehm** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02800/joss.02800/10.21105.joss.02800.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02800/joss.02800/10.21105.joss.02800.pdf) :page_facing_up: :point_left:

**Comment by fboehm** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1371/journal.pone.0065848 is OK
- 10.21105/joss.00451 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by fboehm** [comment]
@henrykironde - would you make a tagged release on github and archive that release at a service like zenodo? After you've done that, please report the version number and doi in a comment on this thread. Thanks!!

**Comment by fboehm** [comment]
Also, @henrykironde - please double-check the latest proof. Be sure that the author list is as you want it to be.

**Comment by henrykironde** [comment]
@fboehm version 3.0.1 DOI `10.5281/zenodo.4314115` url https://zenodo.org/record/4314115#.X9G2kS1h2b8

**Comment by fboehm** [comment]
Thank you, @henrykironde !

**Comment by fboehm** [comment]
@whedon set 10.5281/zenodo.4314115 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.4314115" target="_blank">10.5281/zenodo.4314115</a> is the archive.

**Comment by fboehm** [comment]
@whedon set v3.0.1 as version

**Comment by whedon** [comment]
OK. v3.0.1 is the version.

**Comment by fboehm** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
PDF failed to compile for issue #2800 with the following error: 

 /app/vendor/ruby-2.6.6/lib/ruby/2.6.0/psych.rb:456:in `parse': (6bdf28831f89b1c1e9051225/paper/paper.md): did not find expected key while parsing a block mapping at line 2 column 1 (Psych::SyntaxError)
	from /app/vendor/ruby-2.6.6/lib/ruby/2.6.0/psych.rb:456:in `parse_stream'
	from /app/vendor/ruby-2.6.6/lib/ruby/2.6.0/psych.rb:390:in `parse'
	from /app/vendor/ruby-2.6.6/lib/ruby/2.6.0/psych.rb:277:in `load'
	from /app/vendor/ruby-2.6.6/lib/ruby/2.6.0/psych.rb:578:in `block in load_file'
	from /app/vendor/ruby-2.6.6/lib/ruby/2.6.0/psych.rb:577:in `open'
	from /app/vendor/ruby-2.6.6/lib/ruby/2.6.0/psych.rb:577:in `load_file'
	from /app/vendor/bundle/ruby/2.6.0/bundler/gems/whedon-54ca145bf448/lib/whedon.rb:127:in `load_yaml'
	from /app/vendor/bundle/ruby/2.6.0/bundler/gems/whedon-54ca145bf448/lib/whedon.rb:87:in `initialize'
	from /app/vendor/bundle/ruby/2.6.0/bundler/gems/whedon-54ca145bf448/lib/whedon/processor.rb:38:in `new'
	from /app/vendor/bundle/ruby/2.6.0/bundler/gems/whedon-54ca145bf448/lib/whedon/processor.rb:38:in `set_paper'
	from /app/vendor/bundle/ruby/2.6.0/bundler/gems/whedon-54ca145bf448/bin/whedon:58:in `prepare'
	from /app/vendor/bundle/ruby/2.6.0/gems/thor-0.20.3/lib/thor/command.rb:27:in `run'
	from /app/vendor/bundle/ruby/2.6.0/gems/thor-0.20.3/lib/thor/invocation.rb:126:in `invoke_command'
	from /app/vendor/bundle/ruby/2.6.0/gems/thor-0.20.3/lib/thor.rb:387:in `dispatch'
	from /app/vendor/bundle/ruby/2.6.0/gems/thor-0.20.3/lib/thor/base.rb:466:in `start'
	from /app/vendor/bundle/ruby/2.6.0/bundler/gems/whedon-54ca145bf448/bin/whedon:131:in `<top (required)>'
	from /app/vendor/bundle/ruby/2.6.0/bin/whedon:23:in `load'
	from /app/vendor/bundle/ruby/2.6.0/bin/whedon:23:in `<main>'

**Comment by fboehm** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
PDF failed to compile for issue #2800 with the following error: 

 /app/vendor/ruby-2.6.6/lib/ruby/2.6.0/psych.rb:456:in `parse': (541ca4f0b1eb30b0fe08a643/paper/paper.md): did not find expected key while parsing a block mapping at line 2 column 1 (Psych::SyntaxError)
	from /app/vendor/ruby-2.6.6/lib/ruby/2.6.0/psych.rb:456:in `parse_stream'
	from /app/vendor/ruby-2.6.6/lib/ruby/2.6.0/psych.rb:390:in `parse'
	from /app/vendor/ruby-2.6.6/lib/ruby/2.6.0/psych.rb:277:in `load'
	from /app/vendor/ruby-2.6.6/lib/ruby/2.6.0/psych.rb:578:in `block in load_file'
	from /app/vendor/ruby-2.6.6/lib/ruby/2.6.0/psych.rb:577:in `open'
	from /app/vendor/ruby-2.6.6/lib/ruby/2.6.0/psych.rb:577:in `load_file'
	from /app/vendor/bundle/ruby/2.6.0/bundler/gems/whedon-54ca145bf448/lib/whedon.rb:127:in `load_yaml'
	from /app/vendor/bundle/ruby/2.6.0/bundler/gems/whedon-54ca145bf448/lib/whedon.rb:87:in `initialize'
	from /app/vendor/bundle/ruby/2.6.0/bundler/gems/whedon-54ca145bf448/lib/whedon/processor.rb:38:in `new'
	from /app/vendor/bundle/ruby/2.6.0/bundler/gems/whedon-54ca145bf448/lib/whedon/processor.rb:38:in `set_paper'
	from /app/vendor/bundle/ruby/2.6.0/bundler/gems/whedon-54ca145bf448/bin/whedon:58:in `prepare'
	from /app/vendor/bundle/ruby/2.6.0/gems/thor-0.20.3/lib/thor/command.rb:27:in `run'
	from /app/vendor/bundle/ruby/2.6.0/gems/thor-0.20.3/lib/thor/invocation.rb:126:in `invoke_command'
	from /app/vendor/bundle/ruby/2.6.0/gems/thor-0.20.3/lib/thor.rb:387:in `dispatch'
	from /app/vendor/bundle/ruby/2.6.0/gems/thor-0.20.3/lib/thor/base.rb:466:in `start'
	from /app/vendor/bundle/ruby/2.6.0/bundler/gems/whedon-54ca145bf448/bin/whedon:131:in `<top (required)>'
	from /app/vendor/bundle/ruby/2.6.0/bin/whedon:23:in `load'
	from /app/vendor/bundle/ruby/2.6.0/bin/whedon:23:in `<main>'

**Comment by arfon** [comment]
@fboehm - this PR (https://github.com/ropensci/rdataretriever/pull/283) should fix the paper compilation issue.

**Comment by ethanwhite** [comment]
I just merged @arfon's PR.

**Comment by arfon** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
PDF failed to compile for issue #2800 with the following error: 

 /app/vendor/ruby-2.6.6/lib/ruby/2.6.0/psych.rb:456:in `parse': (6745a8ccdad111558cf0a673/paper/paper.md): did not find expected key while parsing a block mapping at line 2 column 1 (Psych::SyntaxError)
	from /app/vendor/ruby-2.6.6/lib/ruby/2.6.0/psych.rb:456:in `parse_stream'
	from /app/vendor/ruby-2.6.6/lib/ruby/2.6.0/psych.rb:390:in `parse'
	from /app/vendor/ruby-2.6.6/lib/ruby/2.6.0/psych.rb:277:in `load'
	from /app/vendor/ruby-2.6.6/lib/ruby/2.6.0/psych.rb:578:in `block in load_file'
	from /app/vendor/ruby-2.6.6/lib/ruby/2.6.0/psych.rb:577:in `open'
	from /app/vendor/ruby-2.6.6/lib/ruby/2.6.0/psych.rb:577:in `load_file'
	from /app/vendor/bundle/ruby/2.6.0/bundler/gems/whedon-54ca145bf448/lib/whedon.rb:127:in `load_yaml'
	from /app/vendor/bundle/ruby/2.6.0/bundler/gems/whedon-54ca145bf448/lib/whedon.rb:87:in `initialize'
	from /app/vendor/bundle/ruby/2.6.0/bundler/gems/whedon-54ca145bf448/lib/whedon/processor.rb:38:in `new'
	from /app/vendor/bundle/ruby/2.6.0/bundler/gems/whedon-54ca145bf448/lib/whedon/processor.rb:38:in `set_paper'
	from /app/vendor/bundle/ruby/2.6.0/bundler/gems/whedon-54ca145bf448/bin/whedon:58:in `prepare'
	from /app/vendor/bundle/ruby/2.6.0/gems/thor-0.20.3/lib/thor/command.rb:27:in `run'
	from /app/vendor/bundle/ruby/2.6.0/gems/thor-0.20.3/lib/thor/invocation.rb:126:in `invoke_command'
	from /app/vendor/bundle/ruby/2.6.0/gems/thor-0.20.3/lib/thor.rb:387:in `dispatch'
	from /app/vendor/bundle/ruby/2.6.0/gems/thor-0.20.3/lib/thor/base.rb:466:in `start'
	from /app/vendor/bundle/ruby/2.6.0/bundler/gems/whedon-54ca145bf448/bin/whedon:131:in `<top (required)>'
	from /app/vendor/bundle/ruby/2.6.0/bin/whedon:23:in `load'
	from /app/vendor/bundle/ruby/2.6.0/bin/whedon:23:in `<main>'

**Comment by arfon** [comment]
Found one more issue: https://github.com/ropensci/rdataretriever/pull/284

**Comment by ethanwhite** [comment]
Merged.

**Comment by arfon** [comment]
@whedon generate pdf

🤞third time's a charm?

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02800/joss.02800/10.21105.joss.02800.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02800/joss.02800/10.21105.joss.02800.pdf) :page_facing_up: :point_left:

**Comment by fboehm** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1371/journal.pone.0065848 is OK
- 10.21105/joss.00451 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
:wave: @openjournals/joss-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/1974

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/1974, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by danielskatz** [comment]
👋 @henrykironde - sorry that this slipped through the cracks. I'll work on it now

**Comment by danielskatz** [comment]
👋 @henrykironde - please update the metadata in the zenodo archive so that the title matches the title of the paper, and similarly, that the authors match the papers authors (both names and order)

**Comment by danielskatz** [comment]
I've proofread the paper and it looks good - it's ready to accept once the zenodo metadata fixes are made.

**Comment by arfon** [comment]
@henrykironde - just a reminder that we're waiting on your response here.

**Comment by henrykironde** [comment]
Going to get back to you asap

**Comment by henrykironde** [comment]
@arfon I think all issue are set, let me know incase something comes up.

**Comment by arfon** [comment]
@whedon accept deposit=true

**Comment by whedon** [comment]
```
Doing it live! Attempting automated processing of paper acceptance...
```

**Comment by whedon** [comment]
🐦🐦🐦 👉 [Tweet for this paper](https://twitter.com/JOSS_TheOJ/status/1346846334926675968) 👈 🐦🐦🐦

**Comment by whedon** [comment]
🚨🚨🚨 **THIS IS NOT A DRILL, YOU HAVE JUST ACCEPTED A PAPER INTO JOSS!** 🚨🚨🚨

 Here's what you must now do:

0. Check final PDF and Crossref metadata that was deposited :point_right: https://github.com/openjournals/joss-papers/pull/2014
1. Wait a couple of minutes to verify that the paper DOI resolves [https://doi.org/10.21105/joss.02800](https://doi.org/10.21105/joss.02800)
2. If everything looks good, then close this review issue.
3. Party like you just published a paper! 🎉🌈🦄💃👻🤘

 Any issues? Notify your editorial technical team...

**Comment by arfon** [comment]
@RMHogervorst, @jsgalan - many thanks for your reviews here and to @fboehm for editing this submission. JOSS relies upon the volunteer efforts of folks like yourselves - we simply wouldn't be able to do this without you ✨!

@henrykironde - your paper is now accepted and published in JOSS :zap::rocket::boom:

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](https://joss.theoj.org/papers/10.21105/joss.02800/status.svg)](https://doi.org/10.21105/joss.02800)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/joss.02800">
  <img src="https://joss.theoj.org/papers/10.21105/joss.02800/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: https://joss.theoj.org/papers/10.21105/joss.02800/status.svg
   :target: https://doi.org/10.21105/joss.02800
```
This is how it will look in your documentation:

[![DOI](https://joss.theoj.org/papers/10.21105/joss.02800/status.svg)](https://doi.org/10.21105/joss.02800)

**We need your help!**

Journal of Open Source Software is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. You can add your name to the reviewer list here: https://joss.theoj.org/reviewer-signup.html
- Making a small donation to support our running costs here: https://numfocus.org/donate-to-joss
